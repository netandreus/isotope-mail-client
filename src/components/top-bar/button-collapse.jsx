import React from 'react';
import PropTypes from 'prop-types';
import {translate} from 'react-i18next';
import TopBarButton from './top-bar-button';

export const ButtonCollapse = ({t, collapsed, sideBarToggle}) => (
  collapsed && (
    <span isotip={t('topBar.showSideBar')} isotip-position='bottom-start' isotip-size='small'>
      <TopBarButton onClick={sideBarToggle}>menu</TopBarButton>
    </span>)
);

ButtonCollapse.propTypes = {
  collapsed: PropTypes.bool.isRequired,
  sideBarToggle: PropTypes.func.isRequired
};

export default translate()(ButtonCollapse);
