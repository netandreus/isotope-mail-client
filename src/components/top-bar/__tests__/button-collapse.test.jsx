import React from 'react';
import {shallow} from 'enzyme/build/index';
import {ButtonCollapse} from '../button-collapse';

describe('ButtonCollapse component test suite', () => {
  test('Snapshot render, collapsed, should render button', () => {
    // Given
    const props = {t: jest.fn(arg => arg), collapsed: true, sideBarToggle: jest.fn()};
    // When
    const button = shallow(<ButtonCollapse {...props}/>);
    // Then
    expect(button).toMatchSnapshot();
  });
  test('Snapshot render, not collapsed, should NOT render button', () => {
    // Given
    const props = {collapsed: false, sideBarToggle: jest.fn()};
    // When
    const button = shallow(<ButtonCollapse {...props}/>);
    // Then
    expect(button).toMatchSnapshot();
  });
});
