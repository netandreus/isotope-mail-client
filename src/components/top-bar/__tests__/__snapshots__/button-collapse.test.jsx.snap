// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonCollapse component test suite Snapshot render, collapsed, should render button 1`] = `
<span
  isotip="topBar.showSideBar"
  isotip-position="bottom-start"
  isotip-size="small"
>
  <TopBarButton
    className=""
    onClick={[MockFunction]}
  >
    menu
  </TopBarButton>
</span>
`;

exports[`ButtonCollapse component test suite Snapshot render, not collapsed, should NOT render button 1`] = `""`;
